### ISSUE TITLE
MAINT: improve auto-labeling

### ISSUE THREAD

**Issue by tylerjereddy** [MDAnalysis/mdanalysis#3813]
I should mention that the approach to auto-labeling that we currently use for PRs and stuff was rejected upstream and replaced with something that is supposed to work better re: not mutating/removing labels humans already placed on PRs: https://github.com/scipy/scipy/pull/16870

We may want to look into that--I'm almost certain I've seen the auto-labeling doing that kind of annoying thing of removing a human label over here as well.

**Comment by tylerjereddy** [comment]
Original PR was https://github.com/MDAnalysis/mdanalysis/pull/3414

**Comment by IAlibay** [comment]
Yeah it's not really bad, but there are some times the bot gets a bit annoying. I think we agreed this was a starting point, so any improvements would be great.
